src/random/cpu/sample_utils.h

```diff
@@ -279,7 +279,18 @@ class TreeSampler: public BaseSampler<Idx> {
     ResetState(prob);
   }
 
-  Idx Draw() {
+  /* Pick an element from the given distribution and update the tree.
+   *
+   * The parameter decrease is an array of which the length is the number of categories.
+   * Every time an element in the category x is picked, the weight of this category is subtracted
+   * by decrease[x]. It is used to support the case where a category might contains multiple
+   * candidates and decrease[x] is the weight of one candidate of the category x.
+   *
+   * When decrease == nullptr, it means there is only one candidate in each category and will
+   * directly set the weight of the chosen category as 0.
+   *
+   */
+  Idx DrawAndUpdate(const DType *decrease = nullptr) {
```

Comment:
why not just call it `Draw`? this interface should be compatible with the old one right?

Reply:
We need to also modify the interface of `BaseSampler` to make the interface consistent with the super class. But `BaseSampler` does not have template parameter `DType`.